Include/Rover/ScalarView.hpp

```diff
@@ -83,11 +54,21 @@ namespace Rover {
       template<typename GetFwd>
       ScalarView(GetFwd&& get, std::size_t size);
```

Comment:
Can you instead have an overload that takes a ```Trial``` as its argument and a conversion function? This view should be parameterized by the conversion function as well to allow for inlining.

Reply:
The function-based constructor is then redundant.
I think it makes sense to have 2 user-defined constructors:
```
template<typename ConversionFwd>
ScalarView<Trial, Conversion>(const Trial&, ConversionFwd&&); // decay-based deduction
```
and 
```
template<typename OtherScalarView, /* disable copy constructor hiding */>
ScalarView<Trial, Conversion>(OtherScalarView&&); // -> ScalarView<OtherScalarView, void>
```
as the latter allows for a reasonable implicit conversion. It is also useful for erasing `Conversion` without the need for `std::function` as any `ScalarView<Trial, Conversion>` can be converted to `ScalarView<ScalarView<...>, void>`.

Reply:
Or, even better, I think I can move `Trial` type out of the template specification altogether, and instead parameterize it by the type of the scalar (and the conversion function).

Reply:
Nevermind, I won't be able to exclude `Trial` from the parameter list unless I introduce another function pararameter, which is at least as bad as the original. So we are left with two options:
1. Keep it as is.
2. Parameterize with `Trial` and `Conversion`, store a pointer to `Trial` and a copy of `Conversion` as data members.
There is no substantial difference between the two, as both support some form of type erasure, either directly with `std::function` in (1), or using `TrialView` (which in turn uses `std::function` internally) in (2). I feel like dealing with a single template parameter is slightly easier than with 2 though, and it seems to introduce fewer dependencies (no dependency on `TrialView` as a "canonical" trial). So I do not see any clear benefits of changing the approach as of now.
I do have to rename `Get` to `SampleGetter` or something though anyway.

Reply:
```ScalarView``` doesn't need to implement any form of type erasure.

Can you implement ```ScalarView``` as follows?

```
template<typename T, typename C>
class ScalarView {
  public:
    using Trial = T;
    using Converter = C;

    ScalarView(const Trial& trial, Converter converter);

   // Default copy/move/assignment operators;
};
```

If someone wants a "type-erased" ```ScalarView``` they can instantiate it with ```T=TrialView``` and ```C=std::function<...>```.

Reply:
It must not _implement_ it, but it should somehow _support_ it.
In your approach, one technically needs to erase two closures, one inside `TrialView` and one for `Converter`. Right now, the two functions are effectively cascaded before erasing the type of the resulting function. While one can argue whether we should be concerned about performance in type-erased contexts (i.e. Python) at all, there is definitely no benefit in making it even heavier.
Apart from the performance consideration, using `ScalarView` in type-erased contexts clearly gets more complicated.
In deduced contexts (the entire `ScalarView` is always a parameter), the compiled code is expected to be the same, as all that differs is where the reference to `Trial` is captured - whether inside or outside of the lambda.

I am just trying to understand the rationale for making the `Trial` type a part of `ScalarView`. If it is just for prettier constructor syntax, I can create the trial-converter constructor while still parameterizing `ScalarView` with a single function - though it requires a free factory function to include the type of the produced lambda in `ScalarView`'s deduction guide.

Reply:
But the converter is a ```std::function```, which inhibits inlining and has other performance pitfalls. The goal of the above implementation is to take advantage of inlining in the common case. It's not about the constructor, it's about the fact that most conversions will be really dead-simple things that can be trivially inlined. By forcing the use of a ```std::function```, we lose the ability for the compiler to automatically "cascade" these conversions on our behalf.

Reply:
But I am not forcing `std::function`, am I? `m_get` is stored as the original type (which is a lambda in case of `ScalarView`'s only producer, `Model`). The only case where `std::function` is imposed where type erasure is necessary, i.e. the Python exports.